app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt

```diff
@@ -1,7 +1,86 @@
 package ar.com.wolox.android.example.ui.home.news
 
+import ar.com.wolox.android.example.model.New
+import ar.com.wolox.android.example.network.NewsService
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import okhttp3.Headers
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.net.UnknownHostException
 import javax.inject.Inject
 
-// TODO: implement logic
-class NewsPresenter @Inject constructor() : BasePresenter<INewsView>()
+class NewsPresenter @Inject constructor(private val mRetrofitServices: RetrofitServices) : BasePresenter<INewsView>() {
+
+    var nextPage: String? = null
+
+    fun hasMore(): Boolean = nextPage != null
+
+    fun requestNews(paginated: Boolean) {
+        val newsService = mRetrofitServices.getService(NewsService::class.java)
+
+        if (paginated && nextPage == null) {
+            runIfViewAttached(INewsView::hideRefreshing)
+            return
+        }
+
+        val callable = if (!paginated) newsService.getNews() else newsService.getNewsByUrl(url = nextPage!!)
+
+        callable.enqueue(object : Callback<List<New>> {
+            override fun onFailure(call: Call<List<New>>, error: Throwable?) {
+                runIfViewAttached { iNewsView ->
+                    iNewsView.hideRefreshing()
+
+                    if (error is UnknownHostException) {
+                        iNewsView.onNoConnection()
+                    } else {
+                        iNewsView.onUnexpectedError()
+                    }
+                }
+            }
+
+            override fun onResponse(call: Call<List<New>>, response: Response<List<New>>) {
+                if (!response.isSuccessful) {
+                    onFailure(call, null)
+                    return
+                }
+
+                val headers: Headers = response.headers()
+
+                nextPage = null
+                val linkHeader = headers.get("Link")
```

Comment:
@leonplondon I'm not getting why you are wanting to get headers here.

Reply:
To extract pagination url that comes in the response `Link` header